pkg/profilecreator/profilecreator.go

```diff
@@ -231,3 +233,64 @@ func (ghwHandler GHWHandler) CPU() (*cpu.Info, error) {
 func (ghwHandler GHWHandler) Topology() (*topology.Info, error) {
 	return ghw.Topology(ghwHandler.snapShotOptions)
 }
+
+// GetReservedAndIsolatedCPUs returns Reserved and Isolated CPUs
+func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, splitReservedCPUsAcrossNUMA bool) (string, string, error) {
+	if reservedCPUCount < 0 {
+		return "", "", fmt.Errorf("Specified eservered CPU count is negative, please specify it correctly")
+	}
+	topologyInfo, err := ghwHandler.Topology()
+	if err != nil {
+		return "", "", fmt.Errorf("Error obtaining Topology Info from GHW snapshot: %v", err)
+	}
+	reservedCPUSet := cpuset.NewBuilder()
+	isolatedCPUSet := cpuset.NewBuilder()
+	var numaNodeNum int
+	if splitReservedCPUsAcrossNUMA {
+		numaNodeNum = len(topologyInfo.Nodes)
+	} else {
+		numaNodeNum = 1
+	}
+
+	var max = 0
+	reservedPerNuma := reservedCPUCount / numaNodeNum
+	remainder := reservedCPUCount % numaNodeNum
+	for numaID, node := range topologyInfo.Nodes {
+		if splitReservedCPUsAcrossNUMA {
+			if remainder != 0 {
+				max = (numaID+1)*reservedPerNuma + (numaNodeNum - remainder)
+				remainder--
+			} else {
+				max = max + reservedPerNuma
+			}
+		} else {
+			max = reservedCPUCount
+		}
+		//sorting by comparing the first logical core of processor cores
+		sort.Slice(node.Cores, func(i, j int) bool {
+			//sorting the logical cores a processor
+			sort.Slice(node.Cores[i].LogicalProcessors, func(x, y int) bool {
+				return node.Cores[i].LogicalProcessors[x] < node.Cores[i].LogicalProcessors[y]
+			})
+			sort.Slice(node.Cores[j].LogicalProcessors, func(x, y int) bool {
+				return node.Cores[j].LogicalProcessors[x] < node.Cores[j].LogicalProcessors[y]
+			})
+			return node.Cores[i].LogicalProcessors[0] < node.Cores[j].LogicalProcessors[0]
+		})
+		for _, cores := range node.Cores {
+			for _, core := range cores.LogicalProcessors {
+				if reservedCPUSet.Result().Size() < max {
+					reservedCPUSet.Add(core)
+				} else {
+					isolatedCPUSet.Add(core)
```

Comment:
I wonder if it would be clearer to only pick the reservedCPUSet and to compute the isolated set once, outside the loop, as difference like `isolatedCPUSet = totalCPUSet - reservedCPUSet`

Reply:
Sure, that is no problem. I can do that. I had a version where I did it that way but thought populating both the CPUSets was more readable. Will update.

Reply:
+1